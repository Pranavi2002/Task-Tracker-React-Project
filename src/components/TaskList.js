import React from "react";
import TaskItem from "./TaskItem";
import { AnimatePresence } from "framer-motion";
import { FixedSizeList as List } from "react-window";

export default function TaskList({ tasks, onToggle, onDelete, onEdit }) {
  const Row = ({ index, style }) => (
    <div style={style}>
      <TaskItem task={tasks[index]} onToggle={onToggle} onDelete={onDelete} onEdit={onEdit} />
    </div>
  );

  return (
    <AnimatePresence>
      <List height={400} itemCount={tasks.length} itemSize={80} width="100%">
        {Row}
      </List>
    </AnimatePresence>
  );
}