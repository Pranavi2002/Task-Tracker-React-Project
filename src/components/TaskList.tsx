import React from "react";
import TaskItem from "./TaskItem";
import { FixedSizeList as List, ListChildComponentProps } from "react-window";
import { AnimatePresence } from "framer-motion";
import { Task } from "../hooks/useTasks";

interface TaskListProps {
  tasks: Task[];
  onToggle: (id: number) => void;
  onDelete: (id: number) => void;
  onEdit: (id: number, updatedFields: Partial<Task>) => void;
}

const TaskList: React.FC<TaskListProps> = ({ tasks, onToggle, onDelete, onEdit }) => {
  const Row: React.FC<ListChildComponentProps> = ({ index, style }) => (
    <div style={style}>
      <TaskItem task={tasks[index]} onToggle={onToggle} onDelete={onDelete} onEdit={onEdit} />
    </div>
  );

  return (
    <AnimatePresence>
      <List height={400} itemCount={tasks.length} itemSize={60} width="100%">
        {Row}
      </List>
    </AnimatePresence>
  );
};

export default TaskList;