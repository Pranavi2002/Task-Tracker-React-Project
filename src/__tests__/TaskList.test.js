// src/__tests__/TaskList.test.js
import React from "react";
import { render } from "@testing-library/react";
import TaskList from "../components/TaskList";

// Mock TaskItem to simplify snapshot testing
jest.mock("../components/TaskItem", () => (props) => <div>Mocked TaskItem</div>);

describe("TaskList Snapshot", () => {
  // Sample task list
  const tasks = [
    { id: 1, text: "Task 1", completed: false, dueDate: "" },
    { id: 2, text: "Task 2", completed: true, dueDate: "" },
  ];

  // Mock functions for task actions
  const onToggle = jest.fn();
  const onDelete = jest.fn();
  const onEdit = jest.fn();

  it("renders correctly", () => {
    // Render TaskList with mock data
    const { asFragment } = render(
      <TaskList 
        tasks={tasks} 
        onToggle={onToggle} 
        onDelete={onDelete} 
        onEdit={onEdit} 
      />
    );
    // Snapshot testing
    expect(asFragment()).toMatchSnapshot();
  });
});