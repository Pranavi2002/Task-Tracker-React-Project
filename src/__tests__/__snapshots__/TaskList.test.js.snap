// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    style="position: relative; height: 400px; width: 100%; overflow: auto; will-change: transform; direction: ltr;"
  >
    <div
      style="height: 120px; width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; height: 60px; width: 100%;"
      >
        <div>
          Mocked TaskItem
        </div>
      </div>
      <div
        style="position: absolute; left: 0px; top: 60px; height: 60px; width: 100%;"
      >
        <div>
          Mocked TaskItem
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
